<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PWM_HZ</class>
 <widget class="QMainWindow" name="PWM_HZ">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>963</width>
    <height>425</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>STM32 TIMx Calculator</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../resources/images.qrc">
    <normaloff>:/app8.png</normaloff>:/app8.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string/>
        </property>
        <property name="pixmap">
         <pixmap resource="../resources/images.qrc">:/STM32_Frecuency_calculator.png</pixmap>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTextEdit" name="textEdit">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="html">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Consolas'; font-size:10pt;&quot;&gt;Примечание:&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Consolas'; font-size:10pt;&quot;&gt;Timer clock - частота, от которой тактируется таймер (шина APBx)&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Consolas'; font-size:10pt;&quot;&gt;Prescaler - предделитель (регистр PSC)&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Consolas'; font-size:10pt;&quot;&gt;Counter period - до скольки будет считать счетчик CNT. (регистр ARR)&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Consolas'; font-size:10pt;&quot;&gt;Frecuency - итоговая частота в Гц&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Consolas'; font-size:10pt;&quot;&gt;Не забывайте про ограничения таймера.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Consolas'; font-size:10pt;&quot;&gt;Например на STM32F103C8T6 таймеры 16bit, а на STM32F411CCU6 есть таймеры 32 bit&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Consolas'; font-size:10pt;&quot;&gt; &lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QGridLayout" name="gridLayout">
        <item row="5" column="3">
         <widget class="QLabel" name="label_10">
          <property name="font">
           <font>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="text">
           <string>МГц</string>
          </property>
         </widget>
        </item>
        <item row="2" column="3">
         <widget class="QLabel" name="label_6">
          <property name="font">
           <font>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Выставить на STM32:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="4">
         <widget class="QLineEdit" name="lineEdit_CounterPeriod_2">
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>9999</string>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="2">
         <widget class="QLabel" name="label_2">
          <property name="font">
           <font>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Timer clock (Гц):</string>
          </property>
         </widget>
        </item>
        <item row="4" column="3">
         <widget class="QLabel" name="label_9">
          <property name="font">
           <font>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="text">
           <string>кГц</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLineEdit" name="LineEdit_Prescaler">
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>7200</string>
          </property>
         </widget>
        </item>
        <item row="3" column="3">
         <widget class="QLabel" name="label_8">
          <property name="font">
           <font>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Гц</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QLabel" name="label_5">
          <property name="font">
           <font>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Выставить на STM32:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLineEdit" name="LineEdit_TImerClock">
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>72000000</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QLineEdit" name="lineEdit_CounterPeriod">
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>10000</string>
          </property>
         </widget>
        </item>
        <item row="1" column="4">
         <widget class="QLineEdit" name="LineEdit_Prescaler_2">
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>7199</string>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="2">
         <widget class="QLabel" name="label_3">
          <property name="font">
           <font>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Prescaler:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0" colspan="2">
         <widget class="QLabel" name="label_4">
          <property name="font">
           <font>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Counter period:</string>
          </property>
         </widget>
        </item>
        <item row="3" column="2">
         <widget class="QLineEdit" name="lineEdit_FrequencyHz">
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>1</string>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="4" column="2">
         <widget class="QLineEdit" name="lineEdit_FrequencykHz">
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>0.001</string>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="5" column="2">
         <widget class="QLineEdit" name="lineEdit_FrequencyMHz">
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>11</pointsize>
           </font>
          </property>
          <property name="text">
           <string>1e-6</string>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <widget class="QLabel" name="label_7">
          <property name="font">
           <font>
            <pointsize>10</pointsize>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Frequency:</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>963</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources>
  <include location="../resources/images.qrc"/>
 </resources>
 <connections/>
</ui>
